### PR TITLE
[MERGE][REF] *: update owl to version 2 with fine-grained reactivity

### PR DESCRIPTION
**Pull request by jpp-odoo**
This PR updates owl to version 2 with fine-grained reactivity, and adapts the code in Odoo accordingly.

Co-authored-by: Géry Debongnie [ged@odoo.com](mailto:ged@odoo.com)

### CODE REVIEW

File: addons/web/static/src/legacy/js/fields/field_wrapper.js [hunk 1]
```diff
@@ -178,7 +178,8 @@ odoo.define('web.FieldWrapper', function (require) {
                 const fieldsInfo = this.props.record.fieldsInfo[viewType];
                 fieldsInfo[this.props.fieldName].modifiersValue = modifiers || {};
             }
-            this.componentRef.comp.props = this.props;
+            // OWL-NEW-RENDERING
```
Comment (ged-odoo) [inline addons/web/static/src/legacy/js/fields/field_wrapper.js:181]: remoeve

File: addons/web/static/src/search/pager_hook.js [hunk 2]
```diff
@@ -23,11 +23,15 @@ const { useEnv, useChildSubEnv } = owl;
  */
 export function usePager(getProps) {
     const env = useEnv();
+    const pagerState = useState(getProps());
```
Comment (ged-odoo) [inline addons/web/static/src/search/pager_hook.js:26]: replace usestate by reactive

Comment (ged-odoo) [reply addons/web/static/src/search/pager_hook.js:26]: nevermind

File: addons/web/static/src/search/pager_hook.js [hunk 3]
```diff
@@ -23,11 +23,15 @@ const { useEnv, useChildSubEnv } = owl;
...
 
     useChildSubEnv({
         config: {
             ...env.config,
-            getPagerProps: getProps,
+            pagerProps: pagerState,
         },
     });
+    onWillRender(() => {
```
Comment (ged-odoo) [inline addons/web/static/src/search/pager_hook.js:34]: check this

File: addons/mail/static/src/widgets/form_renderer/form_renderer.js [hunk 4]
```diff
@@ -155,6 +155,9 @@ FormRenderer.include({
      * @override
      */
     _updateView() {
+        if (this._hasChatter()) {
```
Comment (ged-odoo) [inline addons/mail/static/src/widgets/form_renderer/form_renderer.js:158]: sadfa

File: addons/point_of_sale/static/src/js/Chrome.js [hunk 5]
```diff
@@ -74,7 +74,8 @@ odoo.define('point_of_sale.Chrome', function(require) {
 
             this.previous_touch_y_coordinate = -1;
 
-            useSubEnv({ pos: useState(this.env.pos) });
+            const pos = owl.reactive(this.env.pos, batched(() => this.render(true)))
```
Comment (ged-odoo) [inline addons/point_of_sale/static/src/js/Chrome.js:77]: remove owl.

File: addons/web/static/src/core/dialog/dialog_container.js [hunk 6]
```diff
@@ -22,18 +22,16 @@ export class DialogContainer extends Component {
         });
     }
 }
-DialogContainer.components = { ErrorHandler, NotUpdatable };
```
Comment (ged-odoo) [inline addons/web/static/src/core/dialog/dialog_container.js:25]: maybe move remove notupdatable in a different commit

**Review by seb-odoo — APPROVED** [review]
Thank you for the change. Just checked the mail part, it's good to see this hook being removed.

I'm not sure what's up with the form/chatter diff, but if it works fine for me.

**Comment by jpp-odoo** [comment]
> Thank you for the change. Just checked the mail part, it's good to see this hook being removed.
> 
> I'm not sure what's up with the form/chatter diff, but if it works fine for me.

Hello @seb-odoo,

Thanks for the review, I just added a comment on the code to roughly explain the form/chatter diff :
> The chatter is detached before it's removed on the super._updateView function, this is done to avoid losing the event handlers.

**Comment by aab-odoo** [comment]
robodoo rebase-merge r+
